src/Translator/doc/index.rst

```diff
@@ -0,0 +1,108 @@
+Symfony UX Translator
+=====================
+
+Symfony UX Translator is a Symfony bundle providing the same mechanism as `Symfony Translator`_
+in JavaScript with a TypeScript integration, in Symfony applications. It is part of `the Symfony UX initiative`_.
+
+The `ICU Message Format`_ is also supported.
+
+Installation
+------------
+
+Before you start, make sure you have `Symfony UX configured in your app`_.
+
+Install this bundle using Composer and Symfony Flex:
+
+.. code-block:: terminal
+
+    $ composer require symfony/ux-translator
+
+    # Don't forget to install the JavaScript dependencies as well and compile
+    $ npm install --force
+    $ npm run watch
+
+    # or use yarn
+    $ yarn install --force
+    $ yarn watch
+
+Usage
+-----
+
+When warming up the Symfony cache, all of your translations will be dumped as JavaScript into the ``var/translations/`` directory.
+For a better developer experience, TypeScript types definitions are also generated aside those JavaScript files.
+
+Then, you will be able to import those JavaScript translations in your assets.
+Don't worry about your final bundle size, only the translations you use will be included in your final bundle, thanks to the [tree shaking](https://webpack.js.org/guides/tree-shaking/).
+
+Configuring the default locale and locale fallbacks
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+By default, if the JavaScript translator is not configured, it will:
+- use the value of ``document.documentElement.lang`` or ``en`` (English) as the default locale
+- not use any locale fallbacks
+
+This step is optional, but if you want to configure the default locale and locale fallbacks for the JavaScript translator, you must use the Twig function ``initialize_js_translator()``:
```

Comment:
Am I correct the the main way people SHOULD set up things is by setting the `<html lang="fr">`? If so, do we need to show a different way? I really like that, and I don't really like the `initialize_js_translator()` (right now, it feels like this is necessary and that it's doing some sort of important JavaScript setup... when in fact, none should be needed).

Reply:
Agreed! What about removing twig function entirely to start (can always add later if there is a valid use-case)?

Reply:
I totally agree too, I'm removing it!

However, sometimes you can't use the lang from `<html lang="...">` because it is not always equals `app.request.local` (e.g.: for SEO reasons, where your app's locales does not follow the same format than `<html lang="...">`). 

I will add a new attribute `<html data-symfony-ux-translator-locale="...">` which will takes precedence over `<html lang="...">`.